Homework4/FtpClient/FtpClient.cs

```diff
@@ -0,0 +1,154 @@
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System;
+
+namespace FtpClient;
+
+public class FtpClient
+{
+    private readonly FtpClientSettings _settings;
+    private readonly CancellationTokenSource _tokenSource;
+    private readonly CancellationToken _token;
+    
+    private bool _keepRunning = true;
+    private NetworkStream _stream = null!;
+
+    public FtpClient(FtpClientSettings settings, CancellationTokenSource cancellationTokenSource)
+    {
+        _settings = settings;
+        _tokenSource = cancellationTokenSource;
+        _token = _tokenSource.Token;
+    }
+
+    private static void PrintHelp()
+    {
+        var commands = new Dictionary<string, string>
+        {
+            { "help", "Print help" },
+            { "ls PATH", "List all entries in the given path" },
+            { "get FILE", "Get file by name" },
+            { "quit", "Exit the program" }
+        };
+        var sb = new StringBuilder();
+        sb.Append("Available commands:\n");
+        foreach (var (command, description) in commands) sb.Append($"* {command}\t{description}\n");
+        Console.WriteLine(sb.ToString());
+    }
+
+    private static async Task WriteStringToStreamAsync(string str, Stream stream)
+    {
+        var bytes = Encoding.UTF8.GetBytes(str);
+        await stream.WriteAsync(bytes);
+    }
+
+
+    private async Task<List<(string name, bool isDir)>> GetList(string path)
+    {
+        if (string.IsNullOrEmpty(path)) path = ".";
+        await WriteStringToStreamAsync($"1 {path}", _stream);
+
+        var bytes = new byte[65535];
+        var size = await _stream.ReadAsync(bytes, _token);
+        var str = Encoding.UTF8.GetString(bytes, 0, size).Split(" ");
+
+        var result = new List<(string name, bool isDir)>();
+        for (var i = 0; i < int.Parse(str[0]); i++) result.Add((str[2 * i + 1], bool.Parse(str[2 * i + 2])));
+
+        return result;
+    }
+
+    private async Task PrintContents(string path)
+    {
+        var contents = await GetList(path);
+
+        var sb = new StringBuilder();
```

Comment:
```suggestion
        var stringBuilder = new StringBuilder();
```
Не нужно сокращать без крайней необходимости